File: 20200924 Onapsis CTF (EKOPARTY)/Pass The Parcel/pass_the_parcel.php
<?php

// run this shit in linux, windows does not scape characters well lol
exec("7z x pass_the_parcel.zip;mv flag.txt flag0.txt");

$i = 0;
while (true) {
    $arre = explode(chr(10), file_get_contents("flag$i.txt"));
    if (count($arre) != 3) {
        echo "\nEKO{" . trim(file_get_contents("flag$i.txt")) . "}\n\n";
        exec("rm flag*");
        die;
    }
    $i++;
    $nuevo = "flag$i.zip";
    file_put_contents($nuevo, base64_decode($arre[2]));
    $pass = escapeshellarg($arre[0]); // get rid of the escaping nightmare
    exec("7z x -p$pass $nuevo;mv flag.txt flag$i.txt");
}

File: 20200924 Onapsis CTF (EKOPARTY)/Pass The Parcel/pass_the_parcel.py
import os
import base64
import shlex # to escape shell args

os.system('7z x pass_the_parcel.zip > /dev/null 2>&1;mv flag.txt flag0.txt')
i = 0
while True:
    with open(f'flag{str(i)}.txt', 'rb') as f:
        Lines = f.readlines()

    if len(Lines) != 3:
        os.system(f'rm flag*');
        print('EKO{' + Lines[0].strip().decode() + '}')
        break
    Pass = shlex.quote(Lines[0].strip().decode())  # get rid of the escaping nightmare
    data = Lines[2]

    i += 1
    nuevo = f'flag{str(i)}.zip'
    g = open(nuevo, 'wb')
    g.write(base64.b64decode(data))
    g.close()
    os.system(f'7z x -p{Pass} {nuevo}> /dev/null 2>&1;mv flag.txt flag{str(i)}.txt');
